internal/services/appconfiguration/app_configuration_replica_resource.go

```diff
@@ -0,0 +1,174 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.Resource = ReplicaResource{}
+
+type ReplicaResource struct{}
+
+func (r ReplicaResource) ModelObject() interface{} {
+	return &ReplicaResourceSchema{}
+}
+
+type ReplicaResourceSchema struct {
+	ConfigurationStoreId string `tfschema:"configuration_store_id"`
+	Endpoint             string `tfschema:"endpoint"`
+	Location             string `tfschema:"location"`
+	Name                 string `tfschema:"name"`
+}
+
+func (r ReplicaResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return replicas.ValidateReplicaID
+}
+func (r ReplicaResource) ResourceType() string {
+	return "azurerm_app_configuration_replica"
+}
+func (r ReplicaResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"configuration_store_id": {
+			ForceNew:     true,
+			Required:     true,
+			Type:         pluginsdk.TypeString,
+			ValidateFunc: configurationstores.ValidateConfigurationStoreID,
+		},
+		"location": commonschema.Location(),
+		"name": {
+			ForceNew:     true,
+			Required:     true,
+			Type:         pluginsdk.TypeString,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+	}
+}
+
+func (r ReplicaResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"endpoint": {
+			Computed: true,
+			Type:     pluginsdk.TypeString,
+		},
+	}
+}
+
+func (r ReplicaResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AppConfiguration.ReplicasClient
+
+			var config ReplicaResourceSchema
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			configurationStoreId, err := configurationstores.ParseConfigurationStoreID(config.ConfigurationStoreId)
+			if err != nil {
+				return fmt.Errorf("parsing configuration store id %s: %+v", config.ConfigurationStoreId, err)
+			}
+
+			id := replicas.NewReplicaID(configurationStoreId.SubscriptionId, configurationStoreId.ResourceGroupName, configurationStoreId.ConfigurationStoreName, config.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for the presence of an existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			payload := replicas.Replica{
+				Location: pointer.To(config.Location),
+				Name:     pointer.To(config.Name),
+			}
+
+			// concurrent creation of replicas under one configuration store will fail
+			locks.ByName(id.ConfigurationStoreName, r.ResourceType())
```

Comment:
The API has limitation that we cannot create two replica under one Configuration Store at the same time, here a lock is added.